test/Test.juvix

```diff
@@ -200,26 +283,126 @@ transposeMatrixDimentionsTest : QC.Test :=
+
+findOnEmptyListIsNothing : QC.Test :=
+  QC.mkTest
+    "find: called with empty list is nothing"
+    prop-findWithEmptyList;
+
+findWithAlwaysTrueIsJust : QC.Test :=
+  QC.mkTest
+    "find: always true predicate returns just"
+    prop-findWithAlwaysTrueIsJust;
+
+findWithAlwaysFalseIsNothing : QC.Test :=
+  QC.mkTest
+    "find: always false predicate returns nothing"
+    prop-findWithAlwaysFalseIsNothing;
+
+resultResultErrorApplication : QC.Test :=
+  QC.mkTest
+    "result: result error application"
+    prop-resultErrorApplication;
+
+resultResultOkApplication : QC.Test :=
+  QC.mkTest
+    "result: result ok application"
+    prop-resultOkApplication;
+
+resultIsError : QC.Test :=
+  QC.mkTest "result: isError detects error" prop-resultIsError;
+
+resultIsOk : QC.Test :=
+  QC.mkTest
+    "result: isOk detects ok"
+    prop-resultIsOk;
+
+resultFromError : QC.Test :=
+  QC.mkTest
+    "result: fromError uses default"
+    prop-resultFromErrorDefault;
+
+resultFromOk : QC.Test :=
+  QC.mkTest
+    "result: fromOk uses default"
+    prop-resultFromOkDefault;
+
+resultResultToMaybe : QC.Test :=
+  QC.mkTest
+    "result: resultToMaybe"
+    prop-resultToMaybe;
+
+resultMaybeToResult : QC.Test :=
+  QC.mkTest
+    "result: maybeToResult"
+    prop-maybeToResult;
+
+resultMapError : QC.Test :=
+  QC.mkTest
+    "result: mapError"
+    prop-resultMapError;
+
+resultMapOk : QC.Test :=
+  QC.mkTest
+    "result: mapOk"
+    prop-resultMapOk;
+
 main : IO :=
   readLn
     \ {seed :=
-      QC.runTestsIO
-        100
-        (stringToNat seed)
-        [ partitionTest
-        ; reverseLengthTest
-        ; reverseReverseIdTest
-        ; splitAtRecombineTest
-        ; splitAtLengthTest
-        ; mergeSumLengthsTest
-        ; tailLengthOneLessTest
-        ; equalCompareToEqTest
-        ; zipTest
-        ; zipWithTest
-        ; snocTest
-        ; dropTest
-        ; sortTest "mergeSort" mergeSort
-        ; sortTest "quickSort" quickSort
-        ; transposeMatrixIdTest
-        ; transposeMatrixDimentionsTest
-        ]};
+      let
+        seedNat := stringToNat seed;
+      in QC.runTestsIO
+          100
+          seedNat
+          [ partitionTest
+          ; reverseLengthTest
+          ; reverseReverseIdTest
+          ; splitAtRecombineTest
+          ; splitAtLengthTest
+          ; mergeSumLengthsTest
+          ; tailLengthOneLessTest
+          ; equalCompareToEqTest
+          ; zipTest
+          ; zipWithTest
+          ; snocTest
+          ; dropTest
+          ; sortTest "mergeSort" mergeSort
+          ; sortTest "quickSort" quickSort
+          ; transposeMatrixIdTest
+          ; transposeMatrixDimentionsTest
+          ; findFoundElementSatisfiesPredicate
+          ; findNonExistenceImpliesPredicateFalseForAll
+          ; findConsistentWithSplitAt
+          ; findOnEmptyListIsNothing
+          ; findWithAlwaysTrueIsJust
+          ; findWithAlwaysFalseIsNothing
+          ]
+        >>> QC.runTestsIO
```

Comment:
The tests are split into two here because the result string is too long for the native backend.